src/frame/index.tsx

```diff
@@ -1,28 +1,125 @@
 import { h, Component } from "preact";
-import API from "../api";
 
 type Props = {
-  api: any;
+  apiKey: string;
+  mode: string;
   url: string;
   width?: number;
   height?: number;
+  data?: any;
+  messageKey?: string;
+  messagePayloadType?: string;
+  messagePayloadActionKey?: string;
 };
 
 type State = {};
 
+const encodeURLParams = (url: string, data: any): string => {
+  let params = "";
+  for (const key in data) {
```

Comment:
There's an easier way to encode an object with:
```js
new URLSearchParams(data).toString()
```

Reply:
Good call. Are there any browser compatibility issues?

Reply:
Browser support is pretty solid https://caniuse.com/#search=URLSearchParams

Reply:
👍